src/rdf4cpp/rdf/datatypes/registry/util/StaticFlatMap.hpp

```diff
@@ -0,0 +1,76 @@
+#ifndef RDF4CPP_REGISTRY_UTIL_STATICFLATMAP_HPP
+#define RDF4CPP_REGISTRY_UTIL_STATICFLATMAP_HPP
+
+#include <cstddef>
+#include <algorithm>
+#include <array>
+#include <utility>
+#include <cassert>
+
+namespace rdf4cpp::rdf::datatypes::registry::util {
+
+/**
+ * A map with fixed/static (maximum) capacity that can be used in constexpr contexts
+ *
+ * @tparam Key key type
+ * @tparam Value mapped type
+ * @tparam capacity maximum number of mappings that can be contained in the map
+ */
+template<typename Key, typename Value, size_t capacity>
+struct StaticFlatMap {
+    using key_type = Key;
+    using mapped_type = Value;
+    using value_type = std::pair<Key, Value>;
+    using const_reference = value_type const &;
+    using const_pointer = value_type const *;
+    using size_type = size_t;
+    using difference_type = ptrdiff_t;
+    using const_iterator = value_type const *;
+
+private:
+    using storage_t = std::array<value_type, capacity>;
+    storage_t storage;
+    size_t max_init;
+
+public:
+    constexpr StaticFlatMap(std::initializer_list<std::pair<Key, Value>> const inits) noexcept(std::is_nothrow_copy_constructible_v<value_type>)
+        : max_init{inits.size()} {
```

Comment:
may contain duplicates. just documenting it is fine.

Reply:
good point